src/datepicker/README.md

```diff
@@ -0,0 +1,124 @@
+    initialState={{
+      value: [new Date(2018, 1, 15), new Date(2018, 8, 25)]
+    }}
+    minDate={new Date(2000, 1, 1)}
+    maxDate={new Date(2020, 1, 1)}
+    onSelect={({event, date})=>{console.log(date)}}
+    />
+}
+```
+
+## Exports
+
+* `Calendar`
+* `StatefulDatepicker`
+* `StatefulContainer`
+* `StyledRoot`
+* `StyledCalendarContainer`
+* `StyledHeader`
+* `StyledMonthHeader`
+* `StyledMonth`
+* `StyledWeek`
+* `StyledDay`
+
+## `Calendar` API
+
+  * `excludeDates?: ?Array<Date> = null`
+    * A list of dates to disable.
+  * `filterDate?: ?(day: Date) => boolean = null`
+    * A filter function that is called to check the disabled state of a day. If `false` is returned the day is considered to be disabled.
+  * `highlightedDate?: ?Date = null`
+    * A date to highlight on an initial render when the selected value is not set.
+  * `includeDates?: ?Array<Date> = null`
+    * A list of selectable dates.
+  * `locale?: ?LocaleT = null`
+    * A locale object. See `date-fns` for more details https://github.com/date-fns/date-fns/tree/master/src/locale 
+  * `maxDate?: ?Date = null`
+    * A max date that is selectable.
+  * `minDate?: ?Date = null`
+    * A min date that is selectable.
+  * `monthsShown?: number = 1` (not supported yet)
+    * A number of monthes rendered in the calendar.
+  * `onDayClick?: ({date: Date, event: Event}) => void`
+    * Day's `click` event handler.
+  * `onDayMouseOver?: ({date: Date, event: Event}) => void`
+    * Day's `mouseover` event handler.
+  * `onDayMouseLeave?: ({date: Date, event: Event}) => void`
+    * Day's `mouseleave` event handler.
+  * `onMonthChange?: ({date: Date}) => void`
+    * Event handler that is called when the current rendered month is changed.
+  * `onYearChange?: ({date: Date}) => void`
+    * Event handler that is called when the current rendered month's year is changed.
+  * `onSelect?: ({date: Date}) => void`
+    * * Event handler that is called when a new date is selected.
+  * `peekNextMonth?: boolean = false`
+    * Defines if dates outside of the range of the current month are displayed.
+  * `selected?: ?Date = null`
+    * Currently selected date.
+  * `setActiveState?: boolean => void = null`
+    * A helper handler for disabling a keyboard navigation and keyboard selection through the calendar dates while navigation through the month or year select controls.
+
+## `StatefulDatepicker` API
+
+* All props of `Calendar` component except of `selected`.
+* `initialState: {}`
+  * Initial state of an uncontrolled datepicker component.
+  * `value: ?Date = null`
+    * Selected `Date`. If `isRange` is set, `value` is an array of 2 values.
+  * `highlightedDate: ?Date = initialState.value || new Date()`
+    * 
+* `stateReducer: (type: text, nextState: {}, currentState: {}, e: any) => nextState`
+  A state change handler.
+  * `type` - state change type
+  * `nextState` - a new state changes that will be set
+  * `currentState` - current full state of the component
```

Comment:
Seems like this is kind of a divergence from the rest of baseui in that the stateful version of a component is the same name but just has a Stateful prefix. Should this maybe be exported as `StatefulCalendar`?